### PR TITLE
Upgrade redis client to redis 2

### PR DESCRIPTION
**Pull request by aredridel**

### CODE REVIEW

**Comment by Marsup** [comment]
Seems dupe of #27 but with different changes, maybe you'd want to contribute there ?

File: lib/index.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ internals.Connection.prototype.start = function (callback) {
 
     // Wait for connection
 
-    client.once('connect', function () {
```
Comment (LoicMahieu) [inline lib/index.js]: Is there a reason of this change ?
I can read in [node_redis documentation](https://github.com/NodeRedis/node_redis#user-content-connect) that `ready` event is not called if `client.options.no_ready_check` is set.

**Comment by LoicMahieu** [comment]
Thanks @aredridel for your PR.
I merged #27 because I wanted to discuss about some different changes in your PR, see comments.

File: lib/index.js (hunk 1 above)
Comment (aredridel) [inline lib/index.js]: Ah, true! 

I was getting un-ready connections later in the tests. I suppose enough of that interface is exposed that that could be set, eh?

**Pull request closed without merging**